x/mint/test/mint_test.go

```diff
@@ -0,0 +1,127 @@
+	// block time to increase by 10 days despite that time not actually passing,
+	// which allows this test to not take 1 year to complete
+	cparams := testnode.DefaultParams()
+	cparams.Block.TimeIotaMs = int64(time.Hour.Milliseconds() * 240)
+
+	cctx, _, _ := testnode.NewNetwork(t, cparams, testnode.DefaultTendermintConfig(), testnode.DefaultAppConfig())
+	s.cctx = cctx
+}
+
+// TestTotalSupplyIncreasesOverTime tests that the total supply of tokens
+// increases over time as new blocks are added to the chain.
+func (s *IntegrationTestSuite) TestTotalSupplyIncreasesOverTime() {
+	require := s.Require()
+
+	err := s.cctx.WaitForNextBlock()
+	require.NoError(err)
+	initalSupply := s.GetTotalSupply()
+
+	_, err = s.cctx.WaitForHeight(20)
+	require.NoError(err)
+	laterSupply := s.GetTotalSupply()
+
+	require.True(initalSupply.AmountOf(app.BondDenom).LT(laterSupply.AmountOf(app.BondDenom)))
+}
+
+// TestInitialInflationRate tests that the initial inflation rate is
+// approximately 8% per year.
+func (s *IntegrationTestSuite) TestInitialInflationRate() {
+	require := s.Require()
+
+	oneYear, err := time.ParseDuration(fmt.Sprintf("%vns", minttypes.NanosecondsPerYear))
+	require.NoError(err)
+
+	err = s.cctx.WaitForNextBlock()
+	require.NoError(err)
+	initialSupply := s.GetTotalSupply()
+	initialTimestamp := s.GetTimestamp()
+
+	_, err = s.cctx.WaitForHeight(20)
+	require.NoError(err)
+	laterSupply := s.GetTotalSupply()
+	laterTimestamp := s.GetTimestamp()
+
+	diffSupply := laterSupply.AmountOf(app.BondDenom).Sub(initialSupply.AmountOf(app.BondDenom))
+	diffTime := laterTimestamp.Sub(initialTimestamp)
+
+	projectedAnnualProvisions := diffSupply.Mul(sdktypes.NewInt(oneYear.Nanoseconds())).Quo(sdktypes.NewInt(diffTime.Nanoseconds()))
+	expectedAnnualProvisions := minttypes.InitalInflationRateDec.Mul(sdktypes.NewDecFromBigInt(initialSupply.AmountOf(app.BondDenom).BigInt())).TruncateInt()
+	diffAnnualProvisions := projectedAnnualProvisions.Sub(expectedAnnualProvisions).Abs()
+
+	// Note we use a .01 margin of error because the projected annual provisions
+	// are based on a small block time iwindow.
+	marginOfError := sdktypes.NewDecWithPrec(1, 2) // .01
+	actualError := sdktypes.NewDecFromBigInt(diffAnnualProvisions.BigInt()).Quo(sdktypes.NewDecFromBigInt(initialSupply.AmountOf(app.BondDenom).BigInt()))
+
+	require.True(actualError.LTE(marginOfError))
+}
+
+func (s *IntegrationTestSuite) GetTotalSupply() sdktypes.Coins {
+	require := s.Require()
+
+	// Note: we can also query for total supply via
+	// r1, r2, err := s.cctx.QueryWithData("/cosmos.bank.v1beta1.Query/TotalSupply", nil)
+	res, err := s.cctx.Client.ABCIQuery(
+		context.Background(),
+		"/cosmos.bank.v1beta1.Query/TotalSupply",
+		nil,
+	)
+	require.NoError(err)
+
+	registry := codectypes.NewInterfaceRegistry()
+	cdc := codec.NewProtoCodec(registry)
+
+	var txResp banktypes.QueryTotalSupplyResponse
+	require.NoError(cdc.Unmarshal(res.Response.Value, &txResp))
+
+	return txResp.GetSupply()
+}
```

Comment:
we should be able to to use the query client and the cctx.GrcpClient for this, no?

https://github.com/celestiaorg/cosmos-sdk/blob/390a17ff29110b74355653477174e9ffef9f849c/x/bank/keeper/grpc_query_test.go#L143-L144

in my experience, using the grpc query clients is much more reliable than pinging the rest endpoints, we can query for state at a given height, and imo its just easier. This does work tho!

Reply:
Attempted to use the GRPC query client in https://github.com/celestiaorg/celestia-app/pull/1791/commits/ee4e3b40a572ede8b0c6e48a6833faf4f3aef06f which results in a panic. Can try again later if using the GRPC client is a blocking comment.

Reply:
are we setting up the bank module's query client somewhere in that commit?

Reply:
🤦‍♂️ that's likely it

Reply:
hmm I tried setting it up in https://github.com/celestiaorg/celestia-app/pull/1791/commits/1d86c4fb119856d73f338df75b143474ac6039b1 but still encountered a `panic`.

Tangent: it's not immediately clear to me which methods on the node use the RPC client vs GRPC client. I can continue debugging this if necessary. Are there other integration tests in our codebase that use GRPC query clients?